### DISCUSSION TITLE
why the type in function "void InstrumentFile::open(const FileName &name)" is different from the epoch:type in InstrumentFile.h?

### DISCUSSION THREAD

**Discussion by zdwangjian5** [groops-devs/groops#200] [Q&A]
I am puzzled by the type in function "void InstrumentFile::open(const FileName &name) . why the case type value  in function "void InstrumentFile::open(const FileName &name)" is different from the epoch:type in InstrumentFile.h? Can anybody can tell me about this?Thank you

**Comment by FFFFFG-IGG** [comment]
Hi,
As far as I can understand from information you provided, the problem may direct at this part(in 'fileInstrument.cpp'):
![image](https://github.com/groops-devs/groops/assets/59271400/fbdc1208-0bf5-48f3-a192-9e38e065599f)
if this is your question, the explanation may be that groops's instrument file has not one version: version before and after 20200123. The difference you mentioned is just a judgement of version in 'fileInstrument.cpp', which means if input instrument file is version before(old), then the type will be transferred to up-to-date(20230123).
regards,
Yongzhao Fan

**Reply by tmayerguerr** [reply]
Dear Yongzhao Fan,
Thank you for your support. You are absolutely right. We restructured the code on 2020-01-23 with a new file format (current version 20200123), but the new GROOPS version should still be able to read old version files.
regards
Torsten
